### ISSUE TITLE
Shouldn't class loss use binary crossentropy?

### ISSUE THREAD

**Issue by daquang** [qqwweee/keras-yolo3#58]
I'm reading the YOLOv3 paper, and it says that YOLOv3 uses the binary crossentropy loss; however, when I looked in the code in this repository, I noticed you use squared error. I'm curious as to how the two losses compare.

**Comment by qqwweee** [comment]
You are right. But as far as i know, Darknet C code uses squared error.

**Comment by qqwweee** [comment]
Maybe I was wrong. It seems that Darknet sets update values directly, and the squared cost is just a number which has nothing to do with Backward Propagation. That is, in Darknet, `y = sigmoid(x); delta_x = 1-y;` means you have a loss function `L = -log(y)` then you get`- dL/dx = 1-y`. So in fact, binary cross entropy is used in Darknet and this repo uses squared error.

**Comment by daquang** [comment]
Thanks! I'm trying to understand the YOLO model and following your code has taught me a lot.

I noticed in line 365 of model.py, you have this:

`
 input_shape = K.cast(K.shape(yolo_outputs[0])[1:3] * 32, K.dtype(y_true[0]))
`

I assume the 32 is a hardcoded batch size. Perhaps you can modify the code a little bit so that people can train with different batch sizes.
